cmd/zapi/cmd/index/create.go

```diff
@@ -0,0 +1,97 @@
+package idx
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"strings"
+
+	"github.com/brimsec/zq/cmd/zapi/cmd"
+	"github.com/brimsec/zq/zqd/api"
+	"github.com/brimsec/zq/zql"
+	"github.com/mccanne/charm"
+)
+
+var Create = &charm.Spec{
+	Name:  "create",
+	Usage: "create [options] [-z zql] [ pattern [ pattern ...]]",
+	Short: "create index on a space",
+	Long: `
+"zapi index create" creates index files in a zar archive using one or more indexing
+rules.
+
+A pattern is either a field name or a ":" followed by a zng type name.
+For example, to index the all fields of type port and the field id.orig_h,
+you would run:
+
+	zapi index create id.orig_h:port
+
+Each pattern results in a separate microindex file for each log file found.
+
+For custom indexes, zql can be used instead of a pattern. This
+requires specifying the key and output file name. For example:
+
+    zapi index create -k id.orig_h -o custom -z "count() by _path, id.orig_h | sort id.orig_h"
+`,
+	New: NewCreate,
+}
+
+func init() {
+	cmd.CLI.Add(Create)
+}
+
+type CreateCmd struct {
+	*cmd.Command
+	root       string
+	inputFile  string
+	outputFile string
+	keys       arrayFlag
+	zql        string
+}
+
+func NewCreate(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
+	c := &CreateCmd{Command: parent.(*IndexCmd).Command}
+	f.Var(&c.keys, "k", "key fields (can be specified multiple times)")
+	f.StringVar(&c.inputFile, "i", "", "input file relative to each zar directory ('' means archive log file in the parent of the zar directory)")
```

Comment:
I'm not a fan of "" for the parent file.  If we don't like "_", maybe we can come up with something better?  "."?  "THIS"?  Unix find uses "{}" inside of exec.  I'm gonna noodle on this.

Reply:
I have found it confusing that you need to know about using `_` to do the default behavior you would expect. That said, if we're going to stick with some special character we should probably add a constant somewhere to the code base to make this easier to see/understand.

Reply:
Also should we expect the `_` be sent over the api call as well?